packages/compartment-mapper/src/types/policy.d.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @file Types required for policy enforcement.
+ */
+
+/* eslint-disable no-use-before-define */
+
+import type { SomeObject } from './typescript.ts';
+
+export type PackageNamingKit = {
+  /** true if location is the entry compartment */
+  isEntry?: boolean | undefined;
+  name: string;
+  path: Array<string>;
+};
+
+export type Attenuator<
+  GlobalParams extends [any, ...any[]] = [any, ...any[]],
+  ModuleParams extends [any, ...any[]] = [any, ...any[]],
+> = {
+  attenuateGlobals?: GlobalAttenuatorFn<GlobalParams> | undefined;
+  attenuateModule?:
+    | ModuleAttenuatorFn<ModuleParams, SomeObject, SomeObject>
+    | undefined;
+};
+
+export type GlobalAttenuatorFn<
+  Params extends [any, ...any[]] = [any, ...any[]],
+> = (
+  params: Params,
+  originalObject: Record<PropertyKey, any>,
+  globalThis: Record<PropertyKey, any>,
+) => void;
+
+export type ModuleAttenuatorFn<
+  Params extends [any, ...any[]] = [any, ...any[]],
+  T = SomeObject,
+  U = T,
+> = (params: Params, ns: T) => U;
```

Comment:
I noticed that all of these `X extends [any, ...any[]] = [any, ...any[]]` should be `X extends readonly [any, ...any[]] = [unknown, ...unknown[]]`.

Also, if it makes sense that e.g., a global attenuator function accepts no parameters whatsoever, then that should be `X extends readonly any[] = unknown[]`.

Reply:
I’m going to leave this for a subsequent fix since the recommended fix did not trivially pass and is beyond my ken.

Reply:
I've put this on my List.